acceptance/tests/load_libfacter.rb

```diff
@@ -0,0 +1,16 @@
+test_name 'Ruby can load libfacter'
+
+agents.each do |agent|
+  if agent.platform.variant == 'windows'
+    facter_gem = 'C:/Program Files/Puppet Labs/Puppet/facter/lib/facter.rb'
+    path = '/cygdrive/c/Windows/system32:/cygdrive/c/Windows'
+  else
+    facter_gem = '/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/facter.rb'
+    path = '/usr/local/bin:/bin:/usr/bin'
+  end
+
+  on agent, "env PATH='#{path}:#{agent['privatebindir']}' ruby -rubygems '#{facter_gem}'" do
```

Comment:
Facter doesn't have a dependency on Puppet or Hiera, so I stripped those out mostly to avoid hard-coding more paths that we'll have to change later. I might drop rubygems as well though, since this is mainly testing loading libfacter.